draft-ietf-jsonpath-base.md

```diff
@@ -1307,92 +1186,168 @@ Queries:
+
+#### Examples
+{: unnumbered}
+
+JSON:
+
+    {"k": "v"}
+
+Queries:
+
+| Query | Result | Result Path | Comment |
+| :---: | ------ | :----------: | ------- |
+| `$` | `{"k": "v"}` | `$` | Root node |
+{: title="Root selector examples"}
+
+### Wildcard Selector {#wildcard}
+
+#### Syntax
+{: unnumbered}
+
+The wildcard selector has either the form `[*]` or the shorthand form `.*`.
+
+~~~~ abnf
+wild-selector        = (index-wild-selector /
+                        dot-wild-selector)
+index-wild-selector  = "[" wildcard "]"       ;  asterisk enclosed by brackets
+dot-wild-selector    = "." wildcard           ;  dot followed by asterisk
+wildcard             = "*"
 ~~~~
 
 #### Semantics
 {: unnumbered}
 
-A list selector selects the nodes that are selected by
-the selector entries in the list and yields the concatenation of the
-lists (in the order of the selector entries) of nodes selected by the
-selector entries.
-Note that any node selected in more than one of the selector entries is kept
-as many times in the nodelist.
+A `wild-selector` selects the nodes of all children of an object or array.
 
-To be valid, integer values in the `element-index` and `slice-index`
-components MUST be in the I-JSON {{-i-json}} range of exact values, see
-{{synsem-overview}}.
+Applying the `wild-selector` to a primitive JSON value (that is,
+a number, a string, `true`, `false`, or `null`) selects no node.
 
+The two formats, `index-wild-selector` and `dot-wild-selector`,
+are functionally identical and may be used interchangeably.
 
 #### Examples
 {: unnumbered}
 
 JSON:
 
-    ["a", "b", "c", "d", "e", "f", "g", {"n": "v", "o": "w", "p": "x"}]
+    {
+      "o": {"j": 1, "k": 2},
+      "a": [5, 3]
+    }
 
 Queries:
 
 | Query | Result | Result Paths | Comment |
 | :---: | ------ | :----------: | ------- |
-| `$[0, 3]` | `"a"` <br> `"d"` | `$[0]` <br> `$[3]` | Indices |
-| `$[0:2, 5]` | `"a"` <br> `"b"` <br> `"f"` | `$[0]` <br> `$[1]` <br> `$[5]` | Slice and index |
-| `$[0, 0]` | `"a"` <br> `"a"` | `$[0]` <br> `$[0]` | Duplicated entries |
-| `$[7]["n", "p"]` | `"v"` <br> `"x"` | `$[7]['n']` <br> `$[7]['p']` | Dot child |
-| `$[? @ <= "b" || @ >= "g", 2]` | `"a"` <br> `"b"` <br> `"g"` <br> `"c"` | `$[0]` <br> `$[1]` <br> `$[6]` <br> `$[2]` | Filter and index |
-{: title="List selector examples"}
+| `$.o[*]` | `1` <br> `2` | `$['o']['j']` <br> `$['o']['k']` | Object values      |
+| `$.o[*]` | `2` <br> `1` | `$['o']['k']` <br> `$['o']['j']` | Alternative result |
+| `$.a[*]` | `5` <br> `3` | `$['a'][0]` <br> `$['a'][1]`     | Array members      |
+{: title="Index wildcard selector examples"}
+
+### Child Selector
+
+#### Syntax
+{: unnumbered}
+
+The child selector has the form `[<__PICKERS__>]` where `<__PICKERS__>` is a comma-delimited
+collection of one or more __PICKERS__.
+Each __PICKER__ is defined below.
+
+~~~~ abnf
+child-selector  = "[" S list-entry 1*(S "," S list-entry) S "]"
+
+list-entry     =  ( quoted-member-name /
```

Comment:
`list-entry` should be tied to __PICKER__ so that the English supports the ABNF. For instance:
```suggestion
PICKER     =  ( quoted-member-name /
```
(where `PICKER` will be replaced by whatever term we choose for __PICKER__).